openapi-load/src/main/kotlin/pro/bilous/difhub/load/ModelLoader.kt

```diff
@@ -34,24 +35,45 @@ class ModelLoader(private val defLoader: DefLoader) : IModelLoader {
 	}
 
 	override fun loadModels(reference: String): List<Model>? {
-		val text = loadString(reference)
+		val text = loadString(reference, null)
 		return if (text.isNullOrEmpty()) null else Json.mapper().readValue<List<Model>>(text)
 	}
 
 	private inline fun <reified T> ObjectMapper.readValue(json: String): T
 			= readValue(json, object : TypeReference<T>(){})
 
-	private fun loadString(reference: String): String? {
+	private fun loadString(reference: String, systemSettings: SystemSettings?): String? {
 		val fixedReference = reference.replace("//", "/")
 
 		if (globalModelCache.containsKey(fixedReference)) {
 			return globalModelCache[fixedReference]
 		}
-		val sourceText = defLoader.load(fixedReference)
+
+		val url = if(systemSettings!=null) {
+			resolveReference(fixedReference, systemSettings)
+		} else {
+			fixedReference
+		}
+
+		val sourceText = defLoader.load(url)
 		if (sourceText == null || sourceText.contains("EntityForbiddenException")) {
 			return null
 		}
 		globalModelCache[fixedReference] = sourceText
 		return sourceText
 	}
+
+	private fun resolveReference(reference: String, systemSettings: SystemSettings) : String {
+
```

Comment:
Please review DifHub.kt file, there is a better place to modify the URL

Reply:
At first I considered the class Config as a class to store system settings (in pair with the class DifHub). But it seems like Config and DifHub work as a factory and couldn't have state. Othewise, I'll review again this possibility